components/raftstore/src/store/worker/region.rs

```diff
@@ -806,9 +798,15 @@ where
                     }
                 }
 
+                let ctx = SnapGenContext {
+                    engine: self.engine.clone(),
+                    mgr: self.mgr.clone(),
+                    router: self.router.clone(),
+                };
                 self.pool.spawn(async move {
                     tikv_alloc::add_thread_memory_accessor();
-                    ctx.handle_gen(
+                    Self::handle_gen(
```

Comment:
why not make it as the member function os `SnapGenContext`

Reply:
How about I do it after you all review the changes, because it will require reordering a few functions.

Reply:
ok